### PR TITLE
retry: Do not retry Load() if file does not exist

### LINKED ISSUE #1523: Lock not found warnings on parallel restic runs
## Output of `restic version`
```
restic 0.8.1
compiled with go1.9.2 on windows/amd64
```

## How did you run restic exactly?
```
restic -o sftp.command="plink -s -l mysftpuser -P 555 -i C:/Users/Myself/mysftpuser_id_rsa.ppk myremotesftplocation.com sftp" --repo sftp:myremotesftplocation.com:/data -p C:\restic\backup_password.txt backup C:\Users\Myself\Documents\SomeDir
```

As described in the title i made two runs, the first run (started a second earlier) worked just fine:
```
password is correct
using parent snapshot a471a9e6
scan [C:\Users\Myself\Documents\SomeDir]
scanned 9 directories, 304 files in 0:00
[0:00] 100.00%  0B/s  1.628 GiB / 1.628 GiB  313 / 313 items  0 errors  ETA 0:00
duration: 0:00, 2076.29MiB/s
snapshot a2522615 saved
```
The second run obviously waited for the lock of the first one, but gave strange messages about that:
```
password is correct
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 252.908174ms: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 402.123616ms: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 909.577925ms: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 2.064778206s: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 3.125126695s: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 4.165555777s: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 3.840376221s: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 8.745851428s: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 17.599426733s: file does not exist
Load(<lock/44ef36d35f>, 0, 0) returned error, retrying after 9.716109781s: file does not exist
using parent snapshot a2522615
scan [C:\Users\Myself\Documents\SomeDir]
scanned 9 directories, 304 files in 0:00
[0:01] 100.00%  1.628 GiB/s  1.628 GiB / 1.628 GiB  313 / 313 items  0 errors  ETA 0:00
duration: 0:01, 1497.99MiB/s
snapshot 8feaf923 saved
```

## What backend/server/service did you use to store the repository?
Linux system (Debian) with SFTP

## Expected behavior
More meaningful message like "waiting for parallel run to be finished" and best would be to include some kind of process id, starting time of the other process and a timeout (and how to solve that on a timeout - i dont know how the locking works...).

## Actual behavior
Messages ("Load...") shown above

## Steps to reproduce the behavior
Use restic on Windows. Use SFTP Backend with plink. Start two parallel runs or restic at the same time for a directory with some files/MBs (else the runs end too fast).

## Do you have any idea what may have caused this?
Looks like the locking is already handled well (the second run waited for the first one to finish + x and then started itself), but the locking handled is presented to the user with not really self explaining messages.

## Do you have an idea how to solve the issue?
Implement an error handler for locking-related things and in that case we have here, present a better message to the user.

## Did restic help you or made you happy in any way?
The answer is 42! ...just kidding ;)
No restic in fact makes me very happy. You wont believe how many free backup solutions i searched and tried (for Linux + Windows). restic is the first one which fulfills my requirements AND works.
To be the killer-backup-app for Windows it would need a nice GUI and native-implemented VSS handling (and logging, and...), but i understand that the basics first need to work very stable and after that one could concentrate on nice extras like the ones i described.

### LINKED ISSUE #4515: restic trying to get lock file that no longer exists
Output of `restic version`
--------------------------
Two devices
wbe-pc: restic 0.16.0 compiled with go1.20.8 on linux/amd64
wbe-tp: restic 0.16.0 compiled with go1.20.8 on linux/amd64

How did you run restic exactly?
-------------------------------

same commands were run on both devices
```bash
#!/bin/bash
export RESTIC_DEBUG=true
export DEBUG_LOG=restic-debug.log
/path/to/restic backup --password-file=/path/to/pass.txt --repo=rclone:r2:my-repo --verbose=2 /path/to/src
/path/to/restic forget --keep-daily=3 --keep-hourly=5 --keep-monthly=4 --keep-tag=forever --keep-weekly=4 --password-file=/path/to/pass.txt --path=/path/to/src --repo=rclone:r2:my-repo --verbose=2
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Cloudflare R2 configured with rclone

Expected behavior
-----------------
Expected all restic commands on both devices to either acquire lock and complete or fail.

Actual behavior
---------------
restic continues retrying to load a lock file that no longer exists.

Steps to reproduce the behavior
-------------------------------
I am able to reproduce the problem by running the above commands in "how did you run restic" section.
But this problem will not happen every time.

Do you have any idea what may have caused this?
-----------------------------------------------
A lock file that exists when restic calls GET /locks/ but is deleted before restic calls /locks/lock-id.
Logs and trace are attached at the end.

Some relevant parts of the logs are included here.

Part 1: wbe-pc saves lock/f813ae713d, does backup, deletes lock
<details>
  <summary>Part 1 log</summary>

   ```log
2023/10/13 13:53:49 logger/log.go:59	logger.(*Backend).List	24	List(lock)
2023/10/13 13:53:49 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	24	------------  HTTP REQUEST -----------
GET /locks/ HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:49 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	24	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 3
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Fri, 13 Oct 2023 18:53:49 GMT
Server: rclone/1.64.0


2023/10/13 13:53:49 rest/rest.go:384	rest.(*Backend).listv2	24	parsing API v2 response
2023/10/13 13:53:49 logger/log.go:61	logger.(*Backend).List	24	  list err <nil>
2023/10/13 13:53:49 restic/json.go:25	restic.SaveJSONUnpacked	1	save new blob lock
2023/10/13 13:53:49 logger/log.go:30	logger.(*Backend).Save	1	Save(<lock/f813ae713d>, 178)
2023/10/13 13:53:49 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP REQUEST -----------
POST /locks/f813ae713de99d0a726541f99405657c639c9f53da53075891718a4c8e2c860a HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Content-Length: 178
Accept: application/vnd.x.restic.rest.v2
Content-Type: application/octet-stream
Accept-Encoding: gzip


2023/10/13 13:53:49 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 0
Accept-Ranges: bytes
Date: Fri, 13 Oct 2023 18:53:49 GMT
Server: rclone/1.64.0


2023/10/13 13:53:49 logger/log.go:32	logger.(*Backend).Save	1	  save err <nil>
2023/10/13 13:53:49 repository/repository.go:505	repository.(*Repository).SaveUnpacked	1	blob <lock/f813ae713d> saved
2023/10/13 13:53:49 logger/log.go:59	logger.(*Backend).List	43	List(lock)
2023/10/13 13:53:49 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	43	------------  HTTP REQUEST -----------
GET /locks/ HTTP/1.1
Host: lcalhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:49 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	43	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 89
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Fri, 13 Oct 2023 18:53:49 GMT
Server: rclone/1.64.0


2023/10/13 13:53:49 rest/rest.go:384	rest.(*Backend).listv2	43	parsing API v2 response
2023/10/13 13:53:49 logger/log.go:61	logger.(*Backend).List	43	  list err <nil>
2023/10/13 13:53:49 restic/parallel.go:44	restic.ParallelList.func2	45	worker got file lock/f813ae71
2023/10/13 13:53:49 restic/lock.go:104	main.lockRepository	1	create lock 0xc0000f3880 (exclusive false)
2023/10/13 13:53:49 restic/lock.go:135	main.refreshLocks	49	start
2023/10/13 13:53:49 logger/log.go:59	logger.(*Backend).List	147	List(snapshot)
2023/10/13 13:53:49 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	147	------------  HTTP REQUEST -----------
GET /snapshots/ HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:50 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	147	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 3917
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Fri, 13 Oct 2023 18:53:50 GMT
Server: rclone/1.64.0

2023/10/13 13:53:50 logger/log.go:32	logger.(*Backend).Save	1	  save err <nil>
2023/10/13 13:53:50 cache/file.go:82	cache.(*Cache).Save	1	Save to cache: <snapshot/e8898dcb11>
2023/10/13 13:53:50 repository/repository.go:505	repository.(*Repository).SaveUnpacked	1	blob <snapshot/e8898dcb11> saved
2023/10/13 13:53:50 restic/lock.go:158	main.refreshLocks	49	terminate
2023/10/13 13:53:50 restic/lock.go:146	main.refreshLocks.func1	49	unlocking repository with lock PID 941753 on wbe-pc by wbe (UID 1000, GID 1000)
lock was created at 2023-10-13 13:53:49 (1.605789258s ago)
storage ID f813ae71
2023/10/13 13:53:50 restic/lock.go:222	main.monitorLockRefresh	146	terminate expiry monitoring
2023/10/13 13:53:50 cache/backend.go:37	cache.(*Backend).Remove	49	cache Remove(<lock/f813ae713d>)
2023/10/13 13:53:50 logger/log.go:38	logger.(*Backend).Remove	49	Remove(<lock/f813ae713d>)
2023/10/13 13:53:50 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	49	------------  HTTP REQUEST -----------
DELETE /locks/f813ae713de99d0a726541f99405657c639c9f53da53075891718a4c8e2c860a HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:50 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	49	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Accept-Ranges: bytes
Date: Fri, 13 Oct 2023 18:53:50 GMT
Server: rclone/1.64.0
Content-Length: 0


2023/10/13 13:53:50 logger/log.go:40	logger.(*Backend).Remove	49	  remove err <nil>
2023/10/13 13:53:50 restic/lock.go:300	main.unlockAll	1	unlocking 0 locks
2023/10/13 13:53:50 restic/cleanup.go:87	main.Exit	1	exiting with status code 0

   ```

</details>

Part 2: wbe-tp saves lock/e623a33a, sees lock/f813ae713d when listing locks, but can not access it when calling GET /lock/lock-id
since it has already been deleted by wbe-pc.

<details>
  <summary>Part 2 log</summary>

   ```log
2023/10/13 13:53:49 logger/log.go:59	logger.(*Backend).List	75	List(lock)
2023/10/13 13:53:49 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	75	------------  HTTP REQUEST -----------
GET /locks/ HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:49 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	75	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 89
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Fri, 13 Oct 2023 18:53:49 GMT
Server: rclone/1.64.0


2023/10/13 13:53:49 rest/rest.go:384	rest.(*Backend).listv2	75	parsing API v2 response
2023/10/13 13:53:49 logger/log.go:61	logger.(*Backend).List	75	  list err <nil>
2023/10/13 13:53:49 restic/parallel.go:44	restic.ParallelList.func2	80	worker got file lock/f813ae71
2023/10/13 13:53:49 repository/repository.go:176	repository.(*Repository).LoadUnpacked	80	load lock with id f813ae713de99d0a726541f99405657c639c9f53da53075891718a4c8e2c860a
2023/10/13 13:53:49 cache/file.go:38	cache.(*Cache).load	80	Load(<lock/f813ae713d>, 0, 0) from cache
2023/10/13 13:53:49 cache/backend.go:172	cache.(*Backend).Load	80	error loading <lock/f813ae713d> from cache: cannot be cached
2023/10/13 13:53:49 cache/backend.go:176	cache.(*Backend).Load	80	Load(<lock/f813ae713d>, 0, 0): delegating to backend
2023/10/13 13:53:49 logger/log.go:45	logger.(*Backend).Load	80	Load(<lock/f813ae713d>, length 0, offset 0)
2023/10/13 13:53:49 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	80	------------  HTTP REQUEST -----------
GET /locks/f813ae713de99d0a726541f99405657c639c9f53da53075891718a4c8e2c860a HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Range: bytes=0-


2023/10/13 13:53:50 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	80	------------  HTTP RESPONSE ----------
HTTP/2.0 206 Partial Content
Content-Length: 178
Accept-Ranges: bytes
Content-Range: bytes 0-177/178
Content-Type: application/octet-stream
Date: Fri, 13 Oct 2023 18:53:50 GMT
Server: rclone/1.64.0


2023/10/13 13:53:50 logger/log.go:47	logger.(*Backend).Load	80	  load err <nil>
2023/10/13 13:53:50 restic/json.go:25	restic.SaveJSONUnpacked	1	save new blob lock
2023/10/13 13:53:50 logger/log.go:30	logger.(*Backend).Save	1	Save(<lock/e623a33abd>, 179)
2023/10/13 13:53:50 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP REQUEST -----------
POST /locks/e623a33abd60eba1e5d0fbc28fe815cb7deaa3efca33552590ede9974453ffee HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Content-Length: 179
Accept: application/vnd.x.restic.rest.v2
Content-Type: application/octet-stream
Accept-Encoding: gzip


2023/10/13 13:53:50 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 0
Accept-Ranges: bytes
Date: Fri, 13 Oct 2023 18:53:50 GMT
Server: rclone/1.64.0


2023/10/13 13:53:50 logger/log.go:32	logger.(*Backend).Save	1	  save err <nil>
2023/10/13 13:53:50 repository/repository.go:505	repository.(*Repository).SaveUnpacked	1	blob <lock/e623a33abd> saved
2023/10/13 13:53:50 logger/log.go:59	logger.(*Backend).List	57	List(lock)
2023/10/13 13:53:50 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	57	------------  HTTP REQUEST -----------
GET /locks/ HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:50 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	57	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 176
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Fri, 13 Oct 2023 18:53:50 GMT
Server: rclone/1.64.0


2023/10/13 13:53:50 rest/rest.go:384	rest.(*Backend).listv2	57	parsing API v2 response
2023/10/13 13:53:50 logger/log.go:61	logger.(*Backend).List	57	  list err <nil>
2023/10/13 13:53:50 restic/parallel.go:44	restic.ParallelList.func2	62	worker got file lock/e623a33a
2023/10/13 13:53:50 restic/parallel.go:44	restic.ParallelList.func2	60	worker got file lock/f813ae71
2023/10/13 13:53:50 repository/repository.go:176	repository.(*Repository).LoadUnpacked	60	load lock with id f813ae713de99d0a726541f99405657c639c9f53da53075891718a4c8e2c860a
2023/10/13 13:53:50 cache/file.go:38	cache.(*Cache).load	60	Load(<lock/f813ae713d>, 0, 0) from cache
2023/10/13 13:53:50 cache/backend.go:172	cache.(*Backend).Load	60	error loading <lock/f813ae713d> from cache: cannot be cached
2023/10/13 13:53:50 cache/backend.go:176	cache.(*Backend).Load	60	Load(<lock/f813ae713d>, 0, 0): delegating to backend
2023/10/13 13:53:50 logger/log.go:45	logger.(*Backend).Load	60	Load(<lock/f813ae713d>, length 0, offset 0)
2023/10/13 13:53:50 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	60	------------  HTTP REQUEST -----------
GET /locks/f813ae713de99d0a726541f99405657c639c9f53da53075891718a4c8e2c860a HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Range: bytes=0-


2023/10/13 13:54:31 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	60	------------  HTTP RESPONSE ----------
HTTP/2.0 404 Not Found
Content-Length: 178
Accept-Ranges: bytes
Content-Range: bytes 0-177/178
Content-Type: text/plain; charset=utf-8
Date: Fri, 13 Oct 2023 18:54:31 GMT
Server: rclone/1.64.0
X-Content-Type-Options: nosniff


2023/10/13 13:54:31 debug/round_tripper.go:46	debug.(*eofDetectReader).Close	60	body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n": Close()
github.com/restic/restic/internal/debug.(*eofDetectReader).Close
	github.com/restic/restic/internal/debug/round_tripper.go:46
github.com/restic/restic/internal/backend/rest.(*Backend).openReader
	github.com/restic/restic/internal/backend/rest/rest.go:228
github.com/restic/restic/internal/backend/rest.(*Backend).Load
	github.com/restic/restic/internal/backend/rest/rest.go:176
github.com/restic/restic/internal/backend/sema.(*connectionLimitedBackend).Load
	github.com/restic/restic/internal/backend/sema/backend.go:95
github.com/restic/restic/internal/backend/logger.(*Backend).Load
	github.com/restic/restic/internal/backend/logger/log.go:46
github.com/restic/restic/internal/backend/retry.(*Backend).Load.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:131
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:47
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer.func1
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:18
github.com/cenkalti/backoff/v4.doRetryNotify[...]
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:88
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:61
github.com/cenkalti/backoff/v4.RetryNotify
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:49
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess
	github.com/restic/restic/internal/backend/retry/backend_retry.go:55
github.com/restic/restic/internal/backend/retry.(*Backend).retry
	github.com/restic/restic/internal/backend/retry/backend_retry.go:77
github.com/restic/restic/internal/backend/retry.(*Backend).Load
	github.com/restic/restic/internal/backend/retry/backend_retry.go:129
github.com/restic/restic/internal/cache.(*Backend).Load
	github.com/restic/restic/internal/cache/backend.go:177
github.com/restic/restic/internal/repository.(*Repository).LoadUnpacked
	github.com/restic/restic/internal/repository/repository.go:189
github.com/restic/restic/internal/restic.LoadJSONUnpacked
	github.com/restic/restic/internal/restic/json.go:14
github.com/restic/restic/internal/restic.LoadLock
	github.com/restic/restic/internal/restic/lock.go:371
github.com/restic/restic/internal/restic.ForAllLocks.func1
	github.com/restic/restic/internal/restic/lock.go:433
github.com/restic/restic/internal/restic.ParallelList.func2
	github.com/restic/restic/internal/restic/parallel.go:45
golang.org/x/sync/errgroup.(*Group).Go.func1
	golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75
runtime.goexit
	runtime/asm_amd64.s:1598
2023/10/13 13:54:31 logger/log.go:47	logger.(*Backend).Load	60	  load err <lock/f813ae713d> does not exist
2023/10/13 13:54:31 restic/global.go:288	main.Warnf	60	Load(<lock/f813ae713d>, 0, 0) returned error, retrying after 625.752961ms: <lock/f813ae713d> does not exist

```

</details>

Part 3: wbe-pc starts second restic command but since wbe-tp has not removed lock/e623a33a, the command will eventually fail out.
<details>
  <summary>Part 3 log</summary>

   ```log
   2023/10/13 13:53:52 logger/log.go:59	logger.(*Backend).List	30	List(lock)
2023/10/13 13:53:52 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	30	------------  HTTP REQUEST -----------
GET /locks/ HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:52 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	30	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 89
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Fri, 13 Oct 2023 18:53:52 GMT
Server: rclone/1.64.0


2023/10/13 13:53:52 rest/rest.go:384	rest.(*Backend).listv2	30	parsing API v2 response
2023/10/13 13:53:52 logger/log.go:61	logger.(*Backend).List	30	  list err <nil>
2023/10/13 13:53:52 restic/parallel.go:44	restic.ParallelList.func2	31	worker got file lock/e623a33a
2023/10/13 13:53:52 repository/repository.go:176	repository.(*Repository).LoadUnpacked	31	load lock with id e623a33abd60eba1e5d0fbc28fe815cb7deaa3efca33552590ede9974453ffee
2023/10/13 13:53:52 cache/file.go:38	cache.(*Cache).load	31	Load(<lock/e623a33abd>, 0, 0) from cache
2023/10/13 13:53:52 cache/backend.go:172	cache.(*Backend).Load	31	error loading <lock/e623a33abd> from cache: cannot be cached
2023/10/13 13:53:52 cache/backend.go:176	cache.(*Backend).Load	31	Load(<lock/e623a33abd>, 0, 0): delegating to backend
2023/10/13 13:53:52 logger/log.go:45	logger.(*Backend).Load	31	Load(<lock/e623a33abd>, length 0, offset 0)
2023/10/13 13:53:52 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	31	------------  HTTP REQUEST -----------
GET /locks/e623a33abd60eba1e5d0fbc28fe815cb7deaa3efca33552590ede9974453ffee HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Range: bytes=0-


2023/10/13 13:53:52 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	31	------------  HTTP RESPONSE ----------
HTTP/2.0 206 Partial Content
Content-Length: 179
Accept-Ranges: bytes
Content-Range: bytes 0-178/179
Content-Type: application/octet-stream
Date: Fri, 13 Oct 2023 18:53:52 GMT
Server: rclone/1.64.0


2023/10/13 13:53:52 logger/log.go:47	logger.(*Backend).Load	31	  load err <nil>
2023/10/13 13:53:52 restic/lock.go:79	main.lockRepository	1	repo already locked, retrying in 0s
2023/10/13 13:53:52 restic/lock.go:86	main.lockRepository	1	repo already locked, timeout expired
2023/10/13 13:53:52 logger/log.go:59	logger.(*Backend).List	85	List(lock)
2023/10/13 13:53:52 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	85	------------  HTTP REQUEST -----------
GET /locks/ HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2023/10/13 13:53:52 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	85	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 89
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Fri, 13 Oct 2023 18:53:52 GMT
Server: rclone/1.64.0


2023/10/13 13:53:52 rest/rest.go:384	rest.(*Backend).listv2	85	parsing API v2 response
2023/10/13 13:53:52 logger/log.go:61	logger.(*Backend).List	85	  list err <nil>
2023/10/13 13:53:52 restic/parallel.go:44	restic.ParallelList.func2	90	worker got file lock/e623a33a
2023/10/13 13:53:52 repository/repository.go:176	repository.(*Repository).LoadUnpacked	90	load lock with id e623a33abd60eba1e5d0fbc28fe815cb7deaa3efca33552590ede9974453ffee
2023/10/13 13:53:52 cache/file.go:38	cache.(*Cache).load	90	Load(<lock/e623a33abd>, 0, 0) from cache
2023/10/13 13:53:52 cache/backend.go:172	cache.(*Backend).Load	90	error loading <lock/e623a33abd> from cache: cannot be cached
2023/10/13 13:53:52 cache/backend.go:176	cache.(*Backend).Load	90	Load(<lock/e623a33abd>, 0, 0): delegating to backend
2023/10/13 13:53:52 logger/log.go:45	logger.(*Backend).Load	90	Load(<lock/e623a33abd>, length 0, offset 0)
2023/10/13 13:53:52 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	90	------------  HTTP REQUEST -----------
GET /locks/e623a33abd60eba1e5d0fbc28fe815cb7deaa3efca33552590ede9974453ffee HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Range: bytes=0-


2023/10/13 13:53:53 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	90	------------  HTTP RESPONSE ----------
HTTP/2.0 206 Partial Content
Content-Length: 179
Accept-Ranges: bytes
Content-Range: bytes 0-178/179
Content-Type: application/octet-stream
Date: Fri, 13 Oct 2023 18:53:53 GMT
Server: rclone/1.64.0


2023/10/13 13:53:53 logger/log.go:47	logger.(*Backend).Load	90	  load err <nil>
2023/10/13 13:53:53 restic/lock.go:300	main.unlockAll	1	unlocking 0 locks
2023/10/13 13:53:53 restic/cleanup.go:87	main.Exit	1	exiting with status code 1

   ```

</details>

Do you have an idea how to solve the issue?
-------------------------------------------
I think this problem might be addressed if there is no retry when Load() fails with noExistError.
I might be overlooking other factors. I would appericiate if anyone could tell me if this makes sense or point me in the right direction.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Despite this cahllenge, restic is what got me to try and setup proper backups on all my devices again. Overall, it has been way more fun than frustration compared to other solutions I've tried in the past. Thank you.

Logs
------------------------------------------------------------
[restic-trace.txt](https://github.com/restic/restic/files/12897977/restic-trace.txt)
[restic-debug-wbe-tp.log](https://github.com/restic/restic/files/12898188/restic-debug-wbe-tp.log)
[restic-debug-wbe-pc.log](https://github.com/restic/restic/files/12898214/restic-debug-wbe-pc.log)

### PR DESCRIPTION
**Pull request by awannabeengineer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
It changes the behavior of retry for Load by not retrying if file does not exist.
This will prevent retrying Load in a situation where no matter the number of retries, Load will not succeed. 
ex: loading the lock file placed in repo by another host that has since been deleted.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
fixes #4515
fixes #1523

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added tests for all code changes.
- <s>[ ] I have added documentation for relevant changes (in the manual).</s>
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: internal/backend/retry/backend_retry.go [hunk 1]
```diff
@@ -126,10 +126,16 @@ func (be *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindRe
 // is returned. rd must be closed after use. If an error is returned, the
 // ReadCloser must be nil.
 func (be *Backend) Load(ctx context.Context, h restic.Handle, length int, offset int64, consumer func(rd io.Reader) error) (err error) {
-	return be.retry(ctx, fmt.Sprintf("Load(%v, %v, %v)", h, length, offset),
+	err = be.retry(ctx, fmt.Sprintf("Load(%v, %v, %v)", h, length, offset),
```
Comment (MichaelEischer) [inline internal/backend/retry/backend_retry.go:129]: Please keep the `return` statement on this line.

File: internal/backend/retry/backend_retry.go [hunk 2]
```diff
@@ -126,10 +126,16 @@ func (be *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindRe
...
 		func() error {
-			return be.Backend.Load(ctx, h, length, offset, consumer)
+			var innerError error
+			innerError = be.Backend.Load(ctx, h, length, offset, consumer)
```
Comment (MichaelEischer) [inline internal/backend/retry/backend_retry.go:132]: `err := ...` would also work here as it declares a new variable that shadows the outer `err`.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
I vaguely seem to remember that some (broken) storage services can sometimes return a "not found" error even for files that actually exist. But I didn't find relevant issues during a quick search. So, I guess we can give this PR a try. (Especially, as we'll definitely need this change to properly implement #1141)

For lock files it's somewhat important to retry a few times if files disappear in the meantime (https://github.com/restic/restic/issues/1523#issuecomment-1221352459). But apparently, I've already implemented that as part of https://github.com/restic/restic/blob/495982232cf1af184eac0a97871ef8161e8708ee/internal/restic/lock.go#L166 .
However, it might be a good idea to modify `ForAllLocks` to skip loading a lock file once it has successfully loaded it once. Although, we can defer that to a separate PR.


Regarding this PR, it definitely needs a changelog entry, see the link in the PR template. Please use `Change` as categorization.

**Comment by awannabeengineer** [comment]
Thanks for the feedback! I amended the commit with your suggested changes and added a changelog entry. If separate commits are preferred, please let me know.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**